### ISSUE TITLE
Add `isPostsPage` in GraphQL pages & page query

### ISSUE THREAD

**Issue by MKlblangenois** [wp-graphql/wp-graphql#1351]
When we query the pages from WPGraphQL, we've `isFrontPage` settings but nowhere we can't found which is the archive page defined in reading settings (even in ReadingSettings).

In my case, I use this plugin with Gatsby and I've found two problems with this :

1. When i create page with `createPages`, I'll got an empty page because content isn't available on archive page
2. Sometimes, I need to add content or anything else in ACF in my archive page, but with that, I can't dynamically generate an archive page with content in Gatsby.

Solution : add params `isArchive` like `isFrontPage` in page(s) query.

**Comment by jasonbahl** [comment]
@MKlblangenois can you explain a bit more of your process on how you add ACF fields to archive pages? To my knowledge, post type archive pages aren't a possible location for ACF Field Groups. 

Does this mean you're creating a page, and then filtering behavior in WordPress to have WordPress use your page as an archive instead of using the built-in template hierarchy for archives?

Or something else?

**Comment by MKlblangenois** [comment]
@jasonbahl thanks for your answer.

In ACF, you can add ACF fields to archive page (where `page type` is equal to `posts page`).

> Does this mean you're creating a page, and then filtering behavior in WordPress to have WordPress use your page as an archive instead of using the built-in template hierarchy for archives?

Not really, I use the "Static page" read option for the post archive page so that the client can edit, by default, the title of the default archive (index.php).

But it happens (quite often) that customers want content at the top and/or bottom of the posts page. I could do a detection on the page ID, however, if the user would delete or modify the read option, the compilation on the side of Gatsby would have problem, and in this case, Gatsby generates a page without content since I cannot detect that it is the archive page.

I know that it is possible in WordPress to use [is_home()](https://developer.wordpress.org/reference/functions/is_home/) to determine if it is the posts page, but not knowing how your plugin works, I prefer not to go ahead on the realization (and especially philisophy) of the implementation of a parameter of this type.

I don't know if I have been clear enough, do not hesitate to ask me for additional information.

**Comment by jasonbahl** [comment]
ah! 

Ok, so I see the `Page Type is equal to Posts Page` location. 

I think the `isArchive` threw me off, because there are archive pages for any post type and not a way to assign ACF fields to any archive for any post type, and no UI in core to edit those archive pages. They're just mapped to templates. Of course there are creative ways to provide UI for these archive pages, but nothing "official"

When it comes the the posts page, I think we could add a field similar to `isFrontPage`. Maybe `isPostsPage`?

I think `isArchive` is a bit misleading because of how `archive` term is used throughout WordPress.

**Comment by MKlblangenois** [comment]
Oh, yes, the term "isPostsPage" is better in this case and will amply resolve my issue!

I agree with you, isArchive isn't really the most appropriate term but it was more a suggestion, sorry for the confusion ^^'

**Comment by woodwoerk** [comment]
I was looking for the same feature so in the meantime registered this field manually. I'm no PHP developer, but the following worked for me! For anyone else reading this before the 1.0 release, try adding the following to your `functions.php`:

```php
add_action( 'graphql_register_types', function() {
    register_graphql_field( 'Page', 'isPostsPage', [
        'type' => 'Boolean',
        'description' => __( 'Whether this is the posts page', 'wp-graphql' ),
        'resolve' => function( $page ) {
            return $page->ID == get_option( 'page_for_posts' );
        }
    ] );
} );
```

**Comment by jasonbahl** [comment]
closed by #1370
